src/test/java/http_server/ServerTest.java

```diff
@@ -1,169 +1,150 @@
+    Router router = new MockRouter().getApp();
 
-        return formattedDate;
+    @BeforeEach
+    public void prepFiles() {
+        FileHandler.deleteDirectory("./public/dog");
+        FileHandler.deleteDirectory("./public/cat");
+    }
+    
+    @AfterEach
+    public void cleanUpFiles() {
+        FileHandler.deleteDirectory("./public/dog");
+        FileHandler.deleteDirectory("./public/cat");
     }
-
-    String dateHeader = "Date: " + currentDateTime() + Stringer.crlf;
-    String serverHeader = "Server: JavaServer/0.1" + Stringer.crlf;
-    Router router = new MockRouter().getApp();
 
     @Test
     public void GET_Request_Is_Responded_With_Headers_And_No_Body() {
         String request = "GET /simple_get HTTP/1.1";
-        String responseLine = "HTTP/1.1 200 OK" + Stringer.crlf;
-        String response = responseLine + dateHeader + serverHeader;
-
         MockClientSocket mockClientSocket = new MockClientSocket(request);
         Session session = new Session(mockClientSocket, router);
-
         session.run();
+        String response = mockClientSocket.getSentData();
 
-        assertEquals(response, mockClientSocket.getSentData());
+        assertEquals("200", Parser.getStatusCode(response));
+        assertEquals(true, Parser.getHeaders(response).containsKey("Date"));
+        assertEquals(true, Parser.getHeaders(response).containsKey("Server"));
     }
 
     @Test
     public void HEAD_Request_Is_Responded_With_Headers_Only() {
         String request = "HEAD /simple_get HTTP/1.1";
-        String responseLine = "HTTP/1.1 200 OK" + Stringer.crlf;
-        String response = responseLine + dateHeader + serverHeader;
-
         MockClientSocket mockClientSocket = new MockClientSocket(request);
         Session session = new Session(mockClientSocket, router);
-
         session.run();
+        String response = mockClientSocket.getSentData();
 
-        assertEquals(response, mockClientSocket.getSentData());
+        assertEquals("200", Parser.getStatusCode(response));
+        assertEquals(true, Parser.getHeaders(response).containsKey("Date"));
+        assertEquals(true, Parser.getHeaders(response).containsKey("Server"));
     }
 
 
     @Test
     public void GET_Request_Is_Responded_With_Headers_And_Body() {
         String request = "GET /harry_potter HTTP/1.1";
-        String responseLine = "HTTP/1.1 200 OK" + Stringer.crlf;
-        String contentTypeHeader = "Content-Type: text/plain"  + Stringer.crlf;
-        String contentLengthHeader = "Content-Length: 48" + Stringer.crlf;
-        String body = Stringer.crlf + "Here are all my favorite movies:\n" + "- Harry Potter\n";
-        String response = responseLine + dateHeader + serverHeader + contentTypeHeader + contentLengthHeader + body;
-
         MockClientSocket mockClientSocket = new MockClientSocket(request);
         Session session = new Session(mockClientSocket, router);
-
         session.run();
+        String response = mockClientSocket.getSentData();
```

Comment:
all the tests have the same setup (passed in a request, and run the session) -- extract this all into one function so the tests look cleaner